blueocean-dashboard/src/main/js/components/rest/GenericApi.js

```diff
@@ -0,0 +1,31 @@
+import { capabilityAugmenter, Fetch, FetchFunctions, logging } from '@jenkins-cd/blueocean-core-js';
+
+const logger = logging.logger('io.jenkins.blueocean.dashboard.branch.RestApi');
+
+/**
+ * Generic options needed for all requests
+ * @type {{credentials: string, method: string}}
+ */
+const fetchOptionsCommon = {
+    credentials: 'include',
+    method: 'GET',
+};
+/**
+ * Helper method to clone common fetchOptions
+ * @returns {*} fetchOptions
+ */
+function prepareOptions() {
+    const fetchOptions = Object.assign({}, fetchOptionsCommon);
+    return fetchOptions;
+}
+
+export class GenericApi {
```

Comment:
Why isnt this in core-js, probably as a part of the Fetch class? like Fetch.getHref() seems better to me.